tests/src/test_bitinputarchive.cpp

```diff
@@ -2029,4 +2030,65 @@ TEMPLATE_TEST_CASE( "BitInputArchive: Extracting a non-existing folder from an a
         REQUIRE_THROWS( info.extractFolderTo( testOutDir, folderPath ) );
         REQUIRE( fs::is_empty( testOutDir.path() ) );
     }
+}
+
+TEMPLATE_TEST_CASE( "BitInputArchive: Extract to callback",
+                    "[bitinputarchive]", tstring, buffer_t, stream_t ) {
+    const TestDirectory testDir{ fs::path{ test_archives_dir } / "extraction" / "multiple_files" };
+
+#ifdef BIT7Z_BUILD_FOR_P7ZIP
+    const auto testArchive = GENERATE( as< TestInputFormat >(),
+                                       TestInputFormat{ "7z", BitFormat::SevenZip },
+                                       TestInputFormat{ "iso", BitFormat::Iso },
+                                       TestInputFormat{ "tar", BitFormat::Tar },
+                                       TestInputFormat{ "wim", BitFormat::Wim },
+                                       TestInputFormat{ "zip", BitFormat::Zip } );
+#else
+    const auto testArchive = GENERATE( as< TestInputFormat >(),
+                                       TestInputFormat{ "7z", BitFormat::SevenZip },
+                                       // TestInputFormat{ "iso", BitFormat::Iso }, iso format has reverse ids of the elements? todo
```

Comment:
Didn't really dug into that, but at first look it seemed to me as if iso had different indices for the files, but I would need to double check that, so you can skip this artifact for now.

I will polish the test once the idea is approved